### PR TITLE
Merge dev branch branch-22.10 to main

### PR DESCRIPTION
**Pull request by NvTimLiu**
Prepare for spark-rapids release of v22.10.0

Note:  merge with **create a new merge commit** instead of `squash merge`

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

File: pom.xml [hunk 1]
```diff
@@ -987,7 +1067,7 @@
         <parquet.hadoop.version>1.10.1</parquet.hadoop.version>
         <spark.version.classifier>spark${buildver}</spark.version.classifier>
         <cuda.version>cuda11</cuda.version>
-        <spark-rapids-jni.version>22.08.0</spark-rapids-jni.version>
+        <spark-rapids-jni.version>22.10.0-SNAPSHOT</spark-rapids-jni.version>
```
[the remark this answers is not in the document]
Comment (GaryShen2008) [reply pom.xml:1070]: We haven't released the JNI 22.10.0, so it keeps snapshot, I think.
We'll update it after tomorrow's release.

[the remark this answers is not in the document]
Comment (jlowe) [reply pom.xml:1070]: There's also a couple of pending PRs against branch-22.10, so it seems this PR is a bit premature.  Are these issues being handled with further commits on this PR or via another merge commit PR?

[the remark this answers is not in the document]
Comment (sameerz) [reply pom.xml:1070]: Looks like the only pending PR at this point is https://github.com/NVIDIA/spark-rapids/pull/6755

**Comment by NvTimLiu** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**